election-portal/app/election-breakdown/page.tsx

```diff
@@ -5,25 +5,62 @@ import Head from "next/head";
 import Image from "next/image";
 
 import styles from "./page.module.css";
-import Rightbar from "./modules/rightbar";
 import Banner from "./modules/banner";
 import Canvas from "./modules/canvas";
+import { redirect } from "next/navigation";
+import { Year } from "@/types/Year";
+import { RaceType } from "@/types/RaceType";
+import Rightbar from "@/components/rightbar";
+import { useRouter } from "next/navigation";
 
 export default function Election_Breakdown_Page() {
   const [isBannerVisible, setIsBannerVisible] = useState<boolean>(true);
-  const [drawMode, setDrawMode] = useState<boolean>(false);
+  const [raceType, setRaceType] = useState<RaceType>(RaceType.Presidential);
+  const [year, setYear] = useState<Year>(Year.TwentyTwo);
+  const [showRightbar, setShowRightbar] = useState<boolean>(true);
+  const [drawMode, setDrawMode] = useState<boolean>(true);
+  const router = useRouter();
+
+  const availableYears = [
+    Year.Sixteen,
+    Year.Eighteen,
+    Year.Twenty,
+    Year.TwentyTwo,
+    Year.TwentyFour,
+  ];
+
+  const availibleBreakdowns = [
+    RaceType.Presidential,
+    RaceType.Senate,
+    RaceType.House,
+    RaceType.Gubernatorial,
+  ];
 
   const toggleBanner = () => {
     setIsBannerVisible(!isBannerVisible);
   };
 
-  const stringTest = (test: string) => {
-    console.log(test);
+  const toggleRightbar = () => {
+    setShowRightbar(!showRightbar);
   };
 
-  // If toggleDraw is true then we need to block mouse input to everything except Rightbar
-  const toggleDraw = () => {
+  const toggleYear = (year: Year) => {
+    setYear(year);
+    console.log("switched to year", year);
```

Comment:
Ideally go ahead and get rid of console.logs because it will be better than having to chase them down later! Feel free to leave comments as needed